ush/ufsda/misc_utils.py

```diff
@@ -80,9 +81,25 @@ def create_batch_job(job_config, working_dir, executable, yaml_path):
 cd {working_dir}
 """
         f.write(commands)
-        if scheduler[job_config['machine']] == 'slurm':
-            f.write(f"srun -n $SLURM_NTASKS {executable} {yaml_path}\n")
+        if single_exec:
+            # standalone jedi application
+            if scheduler[job_config['machine']] == 'slurm':
+                f.write(f"srun -n $SLURM_NTASKS {executable} {yaml_path}\n")
+        else:
+            # TODO (Guillaume): Hard coded for soca in a few places, change that
+            scripts_path = os.path.join(job_config['modulepath'], '../', 'scripts')
+            # run the pre/run/post scripts
+            f.write(f"source load_envar.sh\n")
+            f.write(f"{scripts_path}/exgdas_global_marine_analysis_prep.py\n")
+            f.write(f"cd {working_dir}/analysis\n")
+            f.write(f"export APRUN_SOCAANAL=\"{job_config['mpiexec']} {job_config['mpinproc']} {job_config['ntasks']}\"\n")
+            f.write(f"{scripts_path}/exgdas_global_marine_analysis_run.sh\n")
     logging.info(f"Wrote batch submission script to {batch_script}")
+
+    # make the batch script executable
```

Comment:
Why is this needed? Do you not just do `sbatch script.sh`?

Reply:
I want to be able to do `./submit.sh`. Useful when using interactive job or debugging the prep step.

Reply:
so this is to just debug and doesn't submit the job, but just runs like a standard `.sh` script?

Reply:
yes

Reply:
@CoryMartin-NOAA I removed the part that makes the script executable

Reply:
@CoryMartin-NOAA , referenced in issue #100